sat-extension/src/main/java/org/openhab/tools/analysis/report/SummaryReportExecutionListener.java

```diff
@@ -0,0 +1,265 @@
+
+import org.apache.maven.execution.AbstractExecutionListener;
+import org.apache.maven.execution.ExecutionEvent;
+import org.apache.maven.execution.ExecutionListener;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.PluginParameterExpressionEvaluator;
+import org.apache.maven.plugin.descriptor.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.component.annotations.Component;
+import org.codehaus.plexus.component.annotations.Requirement;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluationException;
+import org.codehaus.plexus.logging.Logger;
+
+/**
+ * Listens to Maven executions, delegates events to the default {@link ExecutionListener} and updates HTML summaries
+ * based on SAT plugin configuration values.
+ *
+ * @author Wouter Born - Initial contribution
+ */
+@Component(role = SummaryReportExecutionListener.class)
+public class SummaryReportExecutionListener extends AbstractExecutionListener {
+
+    @Requirement
+    private Logger logger;
+
+    @Requirement
+    private SummaryReportHtmlGenerator summaryReportHtmlGenerator;
+
+    private final ExecutorService executor = Executors.newSingleThreadExecutor();
+
+    /**
+     * The default ExecutionListener
+     */
+    private ExecutionListener listener;
+
+    /**
+     * Maps each directory for which a summary report needs to be generated to a {@link SummaryUpdater}.
+     */
+    private final Map<String, SummaryUpdater> summaryUpdaters = Collections.synchronizedMap(new HashMap<>());
+
+    /**
+     * Updates HTML summaries in a directory based on SAT plugin configuration values.
+     */
+    private class SummaryUpdater {
+        private final String directory;
+        private final SummaryHtmlGeneration htmlGeneration;
+        private final int htmlGenerationPeriod;
+        private Instant lastUpdate = Instant.MIN;
+
+        public SummaryUpdater(String directory, SummaryHtmlGeneration htmlGeneration, int htmlGenerationPeriod) {
+            this.directory = directory;
+            this.htmlGeneration = htmlGeneration;
+            this.htmlGenerationPeriod = htmlGenerationPeriod;
+        }
+
+        private File update() {
+            File latestSummaryReport = summaryReportHtmlGenerator.generateHtmlSummaryByRules(directory);
+            lastUpdate = Instant.now();
+            return latestSummaryReport;
+        }
+
+        public void incrementalUpdate() {
+            executor.submit(() -> {
+                if (htmlGeneration == SummaryHtmlGeneration.CONTINUOUS
+                        || (htmlGeneration == SummaryHtmlGeneration.PERIODIC
+                                && Instant.now().isAfter(lastUpdate.plusSeconds(htmlGenerationPeriod)))) {
+                    File latestSummaryReport = update();
+
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("Updated static code analysis summary report in:");
```

Comment:
what about: 

```java
logger.debug("Updated static code analysis summary report in: {}", latestSummaryReport.toURI());
```

You can then leave out the `isDebugEnabled` check

Reply:
Nevermind its no such logger :-(

Reply:
Yes there are Plexus and Maven loggers instead of SLF4J which also makes it sometimes difficult to reuse code. :worried: